olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -367,6 +367,18 @@ PartitionResponse PartitionsRepository::GetAggregatedTile(
     const client::OlpClientSettings& settings) {
   const auto fetch_option = request.GetFetchOption();
   const auto& tile_key = request.GetTileKey();
+
+  const auto& root_tile_key = tile_key.ChangedLevelBy(-kAggregateQuadTreeDepth);
+  const auto root_tile_here = root_tile_key.ToHereTile();
+
+  NamedMutex mutex(catalog.ToString() + layer + root_tile_here + "Index");
+  std::unique_lock<NamedMutex> lock(mutex, std::defer_lock);
```

Comment:
This is tricky one :-)
GetAggregatedTile is looking in cache for any tree containing it, not only `tile_key.ChangedLevelBy(-4)`.
So we need to block loading of quad trees from root_tile_key.Level() to tile_key.Level(). Not sure this can be done using just string.

Reply:
It's needed. Since this is the tile that will be used for API call. Our goal is to merge the online requests.